proto/types.proto

```diff
@@ -16,11 +16,83 @@ message ExtraData {
 
 // Message to specify network config for a Firecracker VM
 message FirecrackerNetworkInterface {
+	bool AllowMMDS = 1; // Specifies if metadata service should be available on this network interface
+	FirecrackerRateLimiter InRateLimiter = 2; // Specifies a rate limiter for incoming bytes
+	FirecrackerRateLimiter OutRateLimiter = 3; // Specifies a rate limiter for outgoing bytes
+
+	// CNIConfiguration specifies CNI configuration that will be used to generate
+	// a network interface for a Firecracker VM.
+	CNIConfiguration CNIConfig = 4;
+
+	// StaticNetworkConfiguration specifies static configuration parameters for a
+	// Firecracker VM's network interface
+	StaticNetworkConfiguration StaticConfig = 5;
+}
+
+// Message to specify CNI configuration that will be used to
+// generate a network interface for a Firecracker VM
+message CNIConfiguration {
+	// NetworkName is the name of a CNI network (as found in CNI
+	// configuration files) that will be used to generate the
+	// network interface.
+	string NetworkName = 1;
+
+	// IfName corresponds to the CNI_IFNAME parameter that will be
+	// provided to CNI plugins during invocation.
+	string IfName = 2;
```

Comment:
:( I still don't like this name.  Do you still prefer to leave it as `IfName` instead of `InterfaceName`?